src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -0,0 +1,165 @@
+import math
+
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases.project import ProjectEndpoint, ProjectPermission
+from sentry.models import Project
+from sentry.snuba import discover
+from sentry.utils.dates import parse_stats_period
+
+
+def percentile_fn(data, percentile):
+    """
+    Returns the nth percentile from a sorted list
+
+    :param percentile: A value between 0 and 1
+    :param data: Sorted list of values
+    """
+    if len(data) == 0:
+        return None
+    pecentile_idx = len(data) * percentile
+    if pecentile_idx.is_integer():
+        return data[int(pecentile_idx)]
+    else:
+        return data[int(math.ceil(pecentile_idx)) - 1]
+
+
+class DynamicSamplingPermission(ProjectPermission):
+    # ToDo(ahmed): Revisit the permission level for Dynamic Sampling once the requirements are
+    #  better defined
+    scope_map = {"GET": ["project:write", "project:admin"]}
+
+
+class ProjectDynamicSamplingDistributionEndpoint(ProjectEndpoint):
+    permission_classes = (DynamicSamplingPermission,)
+
+    @staticmethod
+    def _get_sample_rates_data(data):
+        distribution_functions = {
+            "min": lambda lst: min(lst) if len(lst) > 0 else None,
```

Comment:
min/max functions have `default` kyword: https://docs.python.org/3/library/functions.html#min
```
In [4]: t = min(a, default=None)

In [5]: print(t)
None
```

So you can refactor:
```suggestion
            "min": lambda lst: min(lst, default=None),
```

Reply:
Nice! will add that